### PR TITLE
[aot] Refactor C-API error tests

### PR DESCRIPTION
**Pull request by jim19930609**
Issue: #

### CODE REVIEW

**Comment by PENGUINLIONG** [comment]
Thank you! ATM @damnkk is helping us adding more behavior checks, to prevent potential conflicts I'm gonna approve this after #6871 .

File: c_api/tests/gtest_fixture.h [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
+}
+
 class CapiTest : public ::testing::Test {
+ public:
+  void CHECK_TAICHI_SUCCESS() {
+    TiError actual = ti_get_last_error(0, nullptr);
+    EXPECT_EQ(actual, TI_ERROR_SUCCESS);
+  }
+
+  void CHECK_TAICHI_ERROR(TiError expected,
```
Comment (ailzhang) [inline c_api/tests/gtest_fixture.h:17]: how about `EXPECT_TAICHI_ERROR`?

File: c_api/tests/gtest_fixture.h [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+inline bool is_error_ignorable(TiError error) {
+  if (error == TI_ERROR_NOT_SUPPORTED)
+    return true;
+
+  return false;
+}
+
 class CapiTest : public ::testing::Test {
+ public:
+  void CHECK_TAICHI_SUCCESS() {
```
Comment (ailzhang) [inline c_api/tests/gtest_fixture.h:12]: how about `ASSERT_TAICHI_SUCCESS`?

File: c_api/tests/c_api_interface_test.cpp [hunk 3]
```diff
@@ -136,16 +136,10 @@ TEST_F(CapiTest, FailMapDeviceOnlyMemory) {
...
-
-    TI_ASSERT(err == TI_ERROR_INVALID_STATE);
-    TI_ASSERT(std::string(err_msg).find("Assertion failure") !=
-              std::string::npos);
-    TI_ASSERT(std::string(err_msg).find("RHI map memory failed") !=
-              std::string::npos);
-
-    ti_set_last_error(TI_ERROR_SUCCESS, nullptr);
+    CHECK_TAICHI_ERROR(TI_ERROR_INVALID_STATE, "Assertion failure",
+                       false /* reset_error */);
```
Comment (ailzhang) [inline c_api/tests/c_api_interface_test.cpp:140]: nit: can we unify the format as `, /*reset_error=*/false` :P ?

Comment (ailzhang) [inline c_api/tests/c_api_interface_test.cpp:140]: nit: can we unify the format as `, /*reset_error=*/false` :P ?

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**